.github/workflows/docker.yml

```diff
@@ -17,28 +17,120 @@ on:
     types: [opened, synchronize, reopened, ready_for_review]
 
 env:
-  FOREST_TEST_IMAGE: forest-image
-  FOREST_TEST_ARCHIVE: forest-image.tar
+  CI: 1
+  CARGO_INCREMENTAL: 0
+  CACHE_TIMEOUT_MINUTES: 5
+  SCRIPT_TIMEOUT_MINUTES: 30
+  AWS_ACCESS_KEY_ID: '${{ secrets.AWS_ACCESS_KEY_ID }}'
+  AWS_SECRET_ACCESS_KEY: '${{ secrets.AWS_SECRET_ACCESS_KEY }}'
+  RUSTC_WRAPPER: sccache
+  CC: sccache clang
+  CXX: sccache clang++
 
 jobs:
-  build-and-push-docker-image:
+  build-ubuntu-2204-amd64:
+    name: Build forest binaries on Ubuntu-22.04-amd64
+    runs-on: ubuntu-22.04
     # Run the job only if the PR is not a draft.
     # This is done to limit the runner cost.
     if: github.event.pull_request.draft == false
+    steps:
+      - name: Configure SCCache variables
+        run: |
+          # External PRs do not have access to 'vars' or 'secrets'.
+          if [[ "${{secrets.AWS_ACCESS_KEY_ID}}" != "" ]]; then
+            echo "SCCACHE_ENDPOINT=${{ vars.SCCACHE_ENDPOINT}}" >> $GITHUB_ENV
+            echo "SCCACHE_BUCKET=${{ vars.SCCACHE_BUCKET}}" >> $GITHUB_ENV
+            echo "SCCACHE_REGION=${{ vars.SCCACHE_REGION}}" >> $GITHUB_ENV
+          fi
+      - run: lscpu
+      - name: Show IP
+        run: curl ifconfig.me
+        continue-on-error: true
+      - name: Checkout Sources
+        uses: actions/checkout@v4
+      - name: Setup sccache
+        uses: mozilla-actions/sccache-action@v0.0.5
+        timeout-minutes: '${{ fromJSON(env.CACHE_TIMEOUT_MINUTES) }}'
+        continue-on-error: true
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: "go.work"
+      - name: Cargo Install
+        env:
+          # To minimize compile times: https://nnethercote.github.io/perf-book/build-configuration.html#minimizing-compile-times
+          RUSTFLAGS: "-C linker=clang -C link-arg=-fuse-ld=lld"
+        run: make install
+      - uses: actions/upload-artifact@v4
+        with:
+          name: 'forest-linux-amd64'
+          path: |
+            ~/.cargo/bin/forest*
+          if-no-files-found: error
+  
+  build-ubuntu-2204-arm64:
+    name: Build forest binaries on Ubuntu-22.04-arm64
+    runs-on: buildjet-4vcpu-ubuntu-2204-arm
+    # Run the job only if the PR is not a draft.
+    # This is done to limit the runner cost.
+    if: github.event.pull_request.draft == false
```

Comment:
I'd imagine such builds to fail seldom. Can we do it only on main? Alternatively, only in the merge queue?

Reply:
Sure. I plan to disable it for PR in a subsequent PR so that it's proved to be working in this PR.

Reply:
Only `arm64`.

Reply:
@LesnyRumcajs  should I just update this PR since the arm64 build has been verified at [9850d89](https://github.com/ChainSafe/forest/pull/4763/commits/9850d8905f50d115118a9d092d61cdd56dbbde24)

Reply:
Yes, please.

Reply:
Done.